### PR TITLE
Ft integrating travis #168438532

### PR DESCRIPTION
**Pull request by victor-abz**
#### What does this PR do?
Integrate test for Travis CI
Remove passwords returned for users

#### Description of Task to be completed
Integrate Travis Test Files and unit tests

#### How should this be manually tested?
By installing the app with
```
npm install
```
then checking the working with Postman.
you can also test the unit code coverage with
```
npm test
```

#### Any background context you want to provide?
On password, We removed the password that was being returned when fetching one user all many users.

#### What are the relevant pivotal tracker stories?
[#168438532]
#### Screenshots (if appropriate)

### CODE REVIEW

File: server/controllers/userAuthController.js [hunk 1]
```diff
@@ -4,7 +4,10 @@ import Db from '../db';
 const userAuthController = {
   createUser: async (req, res) => {
     const results = await new Db().addUser(req.body)
-    return Helper.handleSuccess(res, 201, 'User created successfully', results);
+    const {0: user} = results
```
Comment (njeri-ngigi) [inline server/controllers/userAuthController.js:7]: What does this statement do?

File: server/db/index.js [hunk 2]
```diff
@@ -119,7 +120,7 @@ class Database {
   async createReview(reviewData, sessionId, userData) {
     const {score, remark} = reviewData
     const { fullName}  = userData
-    // const sessionData = await this.findByProp('sessions','sessionId', sessionId)
+    const sessionData = await this.findByProp('sessions','sessionId', sessionId)
     // if(sessionData.length === 0) {
```
Comment (njeri-ngigi) [inline server/db/index.js]: remove commented out code

File: server/test/mocks/db.mocks.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
+import dotenv from 'dotenv';
+import Db from '../../db/'
+dotenv.config();
+const data = {
+    mentor: {
```
Comment (njeri-ngigi) [inline server/test/mocks/db.mocks.js:5]: What's the point of this file? What is the use of data?

File: server/test/mocks/db.mocks.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        password: 'mentor123',
+        address: 'Kigali',
+        bio: 'Passionate about IT',
+        occupation: 'Developer',
+        expertise: 'Node Js, Javascript, PostgreSQL',
+      },
+    makeAdmin: async()=> {
+    await new Db().changeToMentor('admin', '1')
+    },
+    makeMentor: async()=> {
```
Comment (njeri-ngigi) [inline server/test/mocks/db.mocks.js:59]: what is the use of this method, makeMentor and makeAdmin?

File: server/test/mocks/db.mocks.js (hunk 3 above)
Comment (victor-abz) [reply server/test/mocks/db.mocks.js:5]: Its the Data I am using for testing in Db, 
with the before each test I am
```
    await setupdb.dropAll() --// Droping all data in tables
    await setupdb.databaseData(); --// recreating the Table and adding in dummy data to do test on.
```

File: server/test/mocks/db.mocks.js (hunk 4 above)
Comment (victor-abz) [reply server/test/mocks/db.mocks.js:59]: These data in Mocks are for testing only. I needed to create a database and drop them for each test. and I need to be having the users to test...

File: server/db/index.js (hunk 2 above)
Comment (victor-abz) [reply server/db/index.js]: ok

File: server/controllers/userAuthController.js (hunk 1 above)
Comment (victor-abz) [reply server/controllers/userAuthController.js:7]: The below destructure the results of array I get from query to object. The code was misplaced the following is the right format in the next commit.
```
createUser: async (req, res) => {
    const results = await new Db().addUser(req.body)
    const {0: user} = results
    const token = Helper.generateToken(user);
    const data = {message: 'User created successfully', token: token}
    return Helper.handleSuccess(res, 201, 'User created successfully', data);
```

**Pull request merged**